docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Async Processing
+---
+import {Icons} from '/src/components/icons'
+
+:::info
+
+This page covers the Responsive Async Processor: its benefits, its limitations, how to use it, and how to tune it.
+
+The Async Processor is available with Responsive client **v0.24.0** onward.
+:::
+
+## Why use the Async Processor
+Async processing allows Kafka Streams applications to concurrently process records within a single task, thus changing the upper limit of parallelization from the level of a Kafka partition to the level of a record key. 
+
+It works by giving each StreamThread a dedicated thread pool which it uses to process the records passing through an async processor. Records of the same key are still processed in order and all the same consistency and correctness guarantees of Kafka Streams are maintained, including exactly-once semantics.
+
+While async processing can help improve performance of any application with a large keyspace, it was built to shine when processors include a long-running step such as remote state store requests or generic RPCs. The classic solution for achieving high throughput when you are bottlenecked by network latency is to increase the parallelism of the calls to the target system, which is exactly what the async processor does for Kafka Streams. 
+
+This makes the Responsive Async Processor the perfect tool to complement Responsive state stores, meaning you don’t need to sacrifice performance in order to gain the availability and elasticity benefits of remote storage with Kafka Streams!
+
+## Expected Performance Gains
+
+As with all real-world systems, performance depends heavily on the specific use case and parameters. However, in all of our experiments, async processing showed an improvement over regular Kafka Streams when a topology includes a processor which makes a remote call. 
+
+<img
+    alt={"Responsive Asyc Processor Peformancs"}
+    src={require('/static/img/docs/reference/async-processing/async-processing-perf.png').default}
```

Comment:
thanks for putting together this graph -- it really shows how performance is freed from the influence of long blocking calls. I'm sure the async throughput in the 1ms case would actually be even higher except we were running up against the limits of that node/the resource limits. Still a great way to showcase async performance -- if anything it really drives home how async will let you scale to make use of all the resources available and become limited only by hardware, rather than RPC latency!

Reply:
Actually we should say pretty much exactly that -- we sort of hint at it right now but it would be great to really drive home the point that async processing frees up your application to utilize all the available resources, so you can truly scale up your compute independently! 

..or something like that